ext/standard/tests/array/array_pad.phpt

```diff
@@ -84,4 +84,4 @@ array(4) {
   [3]=>
   float(2)
 }
-array_pad(): Argument #2 ($length) must be less than or equal to 1048576
+array_pad(): Argument #2 ($length) is too large because it results in a padding of 1999997 elements, which is larger than the allowed 1048576 elements
```

Comment:
there should be no arbitrary limit as `1048576`, it was introduced 20 years ago in https://github.com/php/php-src/commit/7192e8af1531a1ed5568c415b75796f626a7abb2

Reply:
I thought the same thing; that removing the limit would be the ideal situation (whilst making sure we don't overflow the array length's limit, but that's a way higher limit that can't be hit in practice).
I'm willing to make a PR to do that, but it depends on the maintainers' opinion.

Reply:
We can use `HT_MAX_SIZE` as new limit

Reply:
We should probably limit in terms of total array size rather than padding size. This would also make the error clearer, and simplify the checks, if you do it in the same PR.

Reply:
Sounds good. I'll make a new PR with that approach to replace this one soon-ish.